src/main/java/com/flightstats/hub/dao/aws/DynamoChannelExistenceCheck.java

```diff
@@ -0,0 +1,38 @@
+package com.flightstats.hub.dao.aws;
+
+import com.flightstats.hub.config.DynamoProperties;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DynamoChannelExistenceCheck extends AbstractIdleService {
+    private final DynamoUtils dynamoUtils;
+    private final DynamoProperties dynamoProperties;
+
+    @Inject
+    public DynamoChannelExistenceCheck(DynamoUtils dynamoUtils, DynamoProperties dynamoProperties) {
+        this.dynamoUtils = dynamoUtils;
+        this.dynamoProperties = dynamoProperties;
+    }
+
+    @Override
+    protected void startUp() {
+        initialize();
+    }
+
+    @Override
+    protected void shutDown() {
+    }
+
+    void initialize() {
+        String tableName = dynamoProperties.getChannelConfigTableName();
+
+        if (!dynamoUtils.doesTableExist(tableName)) {
```

Comment:
Being that this can throw if the table was recently created before it being called I wonder if it would be valuable to add a minimal retry mechanism to this doesTableExist check

Reply:
The general theme with this change is that the infrastructure must be in place before hub starts.  If we start adding back in "what if it isn't" logic, then we might start back down the prior path...what if it exists, but the config is wrong, etc.  It feels cleaner to keep responsibility for infrastructure stability outside of the hub itself.